Marlin/GCodes.h

```diff
@@ -19,14 +19,78 @@
 #include "planner.h"
 
 //===========================================================================
+//================================ GCode List ===============================
+//===========================================================================
+
+
+/**
+ * Look here for descriptions of G-codes:
+ *  - http://linuxcnc.org/handbook/gcode/g-code.html
+ *  - http://objects.reprap.org/wiki/Mendel_User_Manual:_RepRapGCodes
+ *
+ *
+ * -----------------
+ * Implemented Codes
+ * -----------------
+ *
+ *
+ * "M" Codes
+ *
+ * M251 - Queries small pneumatics cartridge to retrieve Syringe status
+ * M253 - Queries cartridge to see if 24 volts are present or not.
+          FFF: Returns whether hot end is active
+          Pneumatics: Returns whether solenoid is active
+ * M272 - Set axis steps-per-unit for one or more axes, X, Y, Z, and E using
+ *        the default ball-bar units
+*/
+
+//===========================================================================
 //============================= Public Functions ============================
 //===========================================================================
 
 
+/*
+* M251: I2C Query Syringe Status, currently only valid for Cart 1
+*/
+inline void gcode_M251() { 
+  I2C__GetGpioSwitch(CART1_ADDR); 
+}
+
+/*
+* M253: I2C Query Vsense, currently only valid for cartridges
+*   C - 0 - 1   Cartridge Address (0 or 1)
+*/
+inline void gcode_M253() {
+  uint8_t i2c_address = 0xFF;
+  // Used to see if we've been given arguments, and to warn you through the
+  // serial port if they're not seen.
+  bool hasC;
+
+  // Desired address for peripheral device
+  if (hasC = code_seen('C')) {
+    switch (int(code_value())) {
+      case 0:
+        i2c_address = CART0_ADDR;
+        break;
+      case 1:
+        i2c_address = CART1_ADDR;
+        break;
+        break;
+    }
+  }
+
+  if (!hasC) {
+    SERIAL_ECHOLNPGM("No cartridge address given");
+    return;
+  }
+
+  I2C__GetVoltageSense(i2c_address);
+}
+
 /**
  * M272: Set axis steps-per-unit for one or more axes, X, Y, Z, and E using
- *       the defualt ball-bar units
- *      (Follows the same syntax as G92)
+ *       the default ball-bar units
+ *       (Follows the same syntax as G92)
  */
 inline void gcode_M272(void) {
   for (int8_t i = 0; i < NUM_AXIS; i++) {
```

Comment:
Does this need to be a signed int? Also, the `value` variable in this function should really have a better name.